src/test/integration-tests/integration-tests.ts

```diff
@@ -0,0 +1,172 @@
+            const mockHttpJSONRequest = sandbox.stub(utils, "httpJSONRequest").callThrough().withArgs(sinon.match(requestOptions)).resolves(mockGaiaRecord);
+            // calling the method
+            let raisedError;
+            try {
+                const resolvedAddress = await cruxWalletClient.resolveCurrencyAddressForCruxID(testCruxId.toString(), "btc");
+            } catch (error) {
+                raisedError = error;
+            }
+            // expectations
+            expect(mockHttpJSONRequest.calledOnce).to.be.true;
+            expect(raisedError).to.be.instanceOf(CruxClientError);
+            expect(raisedError["errorCode"]).to.be.equal(PackageErrorCode.GaiaRecordIntegrityFailed);
+        })
+        it("resolving address of a user whose gaiaHub is tampered (malicious gaiaHub info response)", async () => {
+            const requestOptions = {
+                baseUrl: "https://hub.cruxpay.com",
+                url: "/hub_info",
+            };
+            const mockGaiaHubInfoResponse = {"challenge_text":"[\"gaiahub\",\"0\",\"hub.cruxpay.com\",\"blockstack_storage_please_sign\"]","latest_auth_version":"v1","max_file_upload_size_megabytes":20,"read_url_prefix":"https://gaia.cruxpay.com?fakeKey=${console.log('test')}"};
+            const mockHttpJSONRequest = sandbox.stub(utils, "httpJSONRequest").callThrough().withArgs(sinon.match(requestOptions)).resolves(mockGaiaHubInfoResponse);
+            const resolvedAddress = await cruxWalletClient.resolveCurrencyAddressForCruxID(testCruxId.toString(), "btc");
+            // @ts-ignore
+            expect(mockHttpJSONRequest.parent.neverCalledWith(sinon.match({ baseUrl: "https://gaia.cruxpay.com?fakeKey=${console.log('test')}" }))).to.be.true;
+        })
+        it("resolving address of a user whose gaiaHub is tampered (malicious gaiaHub info response 2)", async () => {
+            const requestOptions = {
+                baseUrl: "https://hub.cruxpay.com",
+                url: "/hub_info",
+            };
+            const mockGaiaHubInfoResponse = {"challenge_text":"[\"gaiahub\",\"0\",\"hub.cruxpay.com\",\"blockstack_storage_please_sign\"]","latest_auth_version":"v1","max_file_upload_size_megabytes":20,"read_url_prefix":"https://gaia.cruxpay.com/f@akeHubPath!"};
+            const mockHttpJSONRequest = sandbox.stub(utils, "httpJSONRequest").callThrough().withArgs(sinon.match(requestOptions)).resolves(mockGaiaHubInfoResponse);
+            let raisedError;
+            try {
+                const resolvedAddress = await cruxWalletClient.resolveCurrencyAddressForCruxID(testCruxId.toString(), "btc");
```

Comment:
I think we have rewritten such test cases to using expect syntax instead of having try catch and validating the exception object. 

Please check with Sanchay for the exact syntax.